components/payment-term.jsx

```diff
@@ -143,31 +202,66 @@ export function PaymentTerm ({
 							{/* <br />Save up to 25% when you pay annually */}
 						</div>
 					) : (
-						<div>
-							<span
-								className={largePrice ? 'ncf__payment-term__large-price' : ''}
-							>
-								{option.price}
-							</span>
-							{option.chargeOnText && (
-								<p className="ncf__payment-term__charge-on-text">
-									{option.chargeOnText}
-								</p>
-							)}
-						</div>
+						// this should cover the cases different than annual, quarterly and monthly
+						// for those containing period on option.value, render custom template, for the rest keep legacy render
+						typeof option.value === 'string' && !!(option.value.match(/^(P[0-9]+(M|Y))$/)) ?
+							<div className="ncf__payment-term__description">
+								{nameMap['custom'].price(option.price)}
+								{nameMap['custom'].monthlyPrice(
+									option.monthlyPrice && option.monthlyPrice !== '0' ? Number(option.monthlyPrice) : getMontlyPriceFromPeriod(option.amount, option.currency, option.value)
+								)}
+								{nameMap['custom'].renewsText(getTimeFromPeriod(option.value))}
+							</div>
+							: (
+								<div>
+									<span className={largePrice ? 'ncf__payment-term__large-price' : ''}>
+										{option.price}
+									</span>
+									{option.chargeOnText && (
+										<p className="ncf__payment-term__charge-on-text">
+											{option.chargeOnText}
+										</p>
+									)}
+								</div>
+							)
 					)}
 				</React.Fragment>
 			);
 		};
 
-		const getDisplayName = () => {
-			let displayName = '';
+		const getTermDisplayName = () => {
+
+			const showTrialCopyInTitle =
+				option.isTrial && !isPrintOrBundle && !isEpaper;
+
+			const defaultTitle = (option.name && nameMap[option.name]) ? nameMap[option.name].title : '';
+
+			const title = isFixedTermOffer
+				? `${offerDisplayName} - ${defaultTitle}`
+				: defaultTitle;
+
+			let termDisplayName = '';
 			if (showTrialCopyInTitle) {
 				const termName = option.displayName ? option.displayName : 'Premium Digital';
-				displayName = `Trial: ${termName} - `;
+				termDisplayName = `Trial: ${termName} - `;
 			}
-			const termPeriod = nameMap[option.name] ? title : option.title;
-			return `${displayName}${termPeriod} `;
+
+			const getTermPeriod = () => {
+				// annual, quarterly and monthly
+				if (nameMap[option.name]) {
+					return title;
+				// custom offer with period provided
+				} else if (typeof option.value === 'string' && !!(option.value.match(/^(P[0-9]+(M|Y))$/))) {
```

Comment:
we return in the if statement, no need for else
```suggestion
				}
				if (typeof option.value === 'string' && !!(option.value.match(/^(P[0-9]+(M|Y))$/))) {
```

Reply:
I have replaced that with some helper already present on n-pricing. please check if it works for you f2b7d53a3472b8f84348daefaabc8bb648984e72